packages/obonode/obojobo-chunks-materia/server/route-helpers.js

```diff
@@ -40,19 +39,43 @@ const ltiToolConsumer = baseUrl => ({
 	tool_consumer_instance_url: baseUrl
 })
 
-const widgetLaunchParams = (document, materiaChunkId, user, visitId, isPreview, baseUrl) => {
+const widgetLaunchParams = (document, visit, user, materiaOboNodeId, baseUrl) => {
 	const params = {
-		lis_outcome_service_url: `${baseUrl}/materia-lti-score-passback`,
 		lti_message_type: 'basic-lti-launch-request',
-		lis_result_sourcedid: createLisResultSourcedId(visitId, materiaChunkId),
 		lti_version: 'LTI-1p0',
-		resource_link_id: materiaChunkId
+		// materia will send scores here
+		lis_outcome_service_url: `${baseUrl}/materia-lti-score-passback`,
+		// represents the container for the user's score this widget can set/update
+		lis_result_sourcedid: `${visit.id}__${materiaOboNodeId}`,
+		// unique placement of this widget (take into account the module's unique placement in a course + the widget's place in the module)
+		// we are intentionally not including user id (not part of the placement) or draft_content_id (to allow updates to a module)
+		resource_link_id: `${visit.resource_link_id}__${visit.draft_id}__${materiaOboNodeId}`,
+		custom_obo_node_id: materiaOboNodeId,
+		custom_visit_id: visit.id,
+		custom_draft_id: visit.draft_id,
+		custom_draft_content_id: visit.draft_content_id,
+		custom_passthrough_resource_link_id: visit.resource_link_id
+	}
+
+	// materia currently uses context_id to group scores and attempts
+	// obojobo doesn't support materia as scoreable questions yet, so the key in use here is intended to:
+	// * support materia in content pages
+	// * re lti launch will reset scores/attempts
+	// * browser reload of the window will resume an attempt/score window
+	// a visit id is a representation of: user_id + lti launch + draft_id + draft_content_id
+	const overrideKeys = { context_id: params.resource_link_id }
```

Comment:
```suggestion
	const overrideKeys = { context_id: params.lis_result_sourcedid }
```

Reply:
Right on, this is exactly what I was suggesting in Slack.  the lis_result_sourcedid makes more sense as the 'window' for materia scores/attempts as it'll change when a new visit is generated - in addition to taking into account the placement and document.